contents/swift/welcome-to-swift.md

```diff
@@ -144,9 +146,45 @@ Here are some of the benefits of using `guard` statement over `if-let` statement
 1. Unlike the `if-let` statement, using the `guard` statement causes `past` to remain defined and can be used till the function exits.
 1. While using `if-let` statements can lead to deeply nested `if-let` statements (i.e. pyramid of doom), `guard` statements allow us to have the happy path to be not indented, thereby increasing code readability.
 
-## Data Types
+### Benefit: Handle Post-operations easily
+
+The `defer` key word in Swift provides an easy and safe way to execute some code before leaving current scope. It will be helpful if you need to do post-operations in the function which has many return points.
+
+The following code is an example of using a file:
+```swift
+let fileDescriptor = open(url.path, O_EVTONLY)
+if fileDescriptor == -1 {
+	close(fileDescriptor)
+	return "Failed"
+}
+
+// Use file descriptor
+
+close(fileDescriptor)
+return "Success"
+```
+
+Suppose that you have many failed cases, so you have to close the file for each case, which requires a lot of extra work. However, you can now consolidate all of them to only one `defer` statement:
```

Comment:
As you can see from above, we have to closing the fileDescriptor for every case we consider. This can be problematic when the number of cases start to increase. Instead, we can use the `defer` statement: